File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.Arrays;
import java.util.ArrayList;
import java.util.Collections;
class Solution {
    public int[] solution(int[] arr, int divisor) {
        int[] answer = {};
        ArrayList<Integer> list = new ArrayList<>();
        // 배열 돌면서 나눠 떨어지는 것 확인
        for(int i = 0; i<arr.length; i++){
            if(arr[i]%divisor ==0){
                // 리스트에 넣기
                list.add(arr[i]);
            }
        }
        
        // 리스트 비어 있으면 -1 넣기
        if(list.isEmpty()){
            list.add(-1);
        }
        // 오름차순 정렬하기기
        Collections.sort(list);
        // 배열로 변환해서 보이기
        
        return list.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.kt
class Solution {
    fun solution(arr: IntArray, divisor: Int): IntArray {
        val result = arr.filter { it % divisor == 0 }  // divisor로 나누어 떨어지는 요소 필터링
            .sorted()  // 오름차순으로 정렬
        return if (result.isEmpty()) intArrayOf(-1) else result.toIntArray()  // 비어있으면 [-1], 그렇지 않으면 결과 반환
    }
}
